### PR TITLE
docs: Add breaking changes note to 13.0 release

### LINKED ISSUE #44383: Route reuse strategy throwing RangeError: Maximum call stack size exceeded in Angular 13
### Which @angular/* package(s) are the source of the bug?

router

### Is this a regression?

No

### Description

We have a considerable large angular app, where we use custom route reuse strategy for performance gains.
The route reuse strategy we use was working perfectly fine in angular 12. However, when we upgraded to Angular 13 we have started seeing the following error:

`Error: Uncaught (in promise): RangeError: Maximum call stack size exceeded`

The error happens in this scenario:
- Go to Route A
- Go to Route B
- Go to Route A again(either by hitting browser back, or by going to the route A through href on <a> tag on page B), crash and navigation doesn't happen.

The inifinite loop is happening inside Angular router's code (router.mjs) in RouterStateSnapshot's constructor, that calls into following function:

`function setRouterState(state, node) {
    node.value._routerState = state;
    node.children.forEach(c => setRouterState(state, c));
}`

Here is how we have our custom route reuse strategy:

```
import {
	RouteReuseStrategy,
	DetachedRouteHandle,
	ActivatedRouteSnapshot,
} from "@angular/router";

export class CustomRouteReuseStrategy implements RouteReuseStrategy {
	private handlers: { [key: string]: DetachedRouteHandle } = {};

	constructor() {}

	public shouldDetach(_route: ActivatedRouteSnapshot): boolean {
		return true;
	}

	public store(
		route: ActivatedRouteSnapshot,
		handle: DetachedRouteHandle
	): void {
		this.handlers[route.data.title] = handle;
	}

	public shouldAttach(route: ActivatedRouteSnapshot): boolean {
		return !!this.handlers[route.data.title];
	}

	public retrieve(route: ActivatedRouteSnapshot): DetachedRouteHandle {
		return this.handlers[route.data.title];
	}

	public shouldReuseRoute(
		future: ActivatedRouteSnapshot,
		curr: ActivatedRouteSnapshot
	): boolean {
		const reuse = future.routeConfig === curr.routeConfig;
		return reuse;
	}
}
```

Here is how we add route to Routing Module:

`
	{
		path: "post/:id",
		loadChildren: () =>
			import("path-to-post-module").then(
				(module) => module.PostModule
			),
		data: { title: "post" },
	},`

This may sound like a stack overflow question, but this setup used to work in Angular 12 without any issues. So, I'm not sure what changed in version 13, that's causing this crash. If I remove the route reuse strategy, this error doesn't happen, but obviously we want the route reuse in our app.

### Please provide a link to a minimal reproduction of the bug
https://github.com/ChabuRah/angular-router-error

### Please provide the exception or error you saw

```true
Error: Uncaught (in promise): RangeError: Maximum call stack size exceeded
RangeError: Maximum call stack size exceeded
    at setRouterState (router.mjs:1610)
    at router.mjs:1612
    at Array.forEach (<anonymous>)
    at setRouterState (router.mjs:1612)
    at router.mjs:1612
    at Array.forEach (<anonymous>)
    at setRouterState (router.mjs:1612)
    at router.mjs:1612
    at Array.forEach (<anonymous>)
    at setRouterState (router.mjs:1612)
    at resolvePromise (zone.js:1213)
    at resolvePromise (zone.js:1167)
    at zone.js:1279
    at ZoneDelegate.invokeTask (zone.js:406)
    at Object.onInvokeTask (core.mjs:25463)
    at ZoneDelegate.invokeTask (zone.js:405)
    at Zone.runTask (zone.js:178)
    at drainMicroTaskQueue (zone.js:582)
    at invokeTask (zone.js:491)
    at ZoneTask.invoke (zone.js:476)
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 13.0.3
Node: 14.15.5
Package Manager: npm 6.14.11
OS: linux x64

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1300.3 (cli-only)
@angular-devkit/core         13.0.3 (cli-only)
@angular-devkit/schematics   13.0.3 (cli-only)
@schematics/angular          13.0.3 (cli-only)
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by atscott**
A breaking change in v13 was identified post-release and needs to be noted
in the breaking changes section to ensure developers are aware of it
when upgrading.

Resolves #44383

### CODE REVIEW

**Review by AndrewKushnir — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -158,6 +158,246 @@ A. Singh, Alan Agius, Alex Rickabaugh, Andrew Kushnir, Andrew Scott, Aristeidis
...
+| -- | -- | -- |
+| [c85bcb0c63](https://github.com/angular/angular/commit/c85bcb0c636d65706541d44be62694720f12ad9e) | feat | reference ICU message IDs from their placeholders ([#43534](https://github.com/angular/angular/pull/43534)) |
+### localize
+| Commit | Type | Description |
+| -- | -- | -- |
+| [d3cf222a81](https://github.com/angular/angular/commit/d3cf222a8193bc7ccd9e7ce886846d1f79131627) | feat | support "associated message ids" for placeholders ([#43534](https://github.com/angular/angular/pull/43534)) |
+## Special Thanks
+Alex Rickabaugh, Andrew Kushnir, Andrew Scott, Doug Parker, Dylan Hunn, George Kalpakas, Jessica Janiuk, Joey Perrott, Paul Gschwendtner and Pete Bacon Darwin
+
+
```
Comment (YegorBobrovski) [inline CHANGELOG.md:174]:
```suggestion
```

IMO this block is no more needed since 13.1.0 is already released

Comment (atscott) [reply CHANGELOG.md:174]: They are release notes that were accidentally deleted and should be restored.

**Review by dylhunn — APPROVED** [review]

**Comment by dylhunn** [comment]
Unclear why this was omitted from the changelog. I think we can just go ahead and merge this ASAP if you're OK with it.

**Review by alxhub — APPROVED** [review]

**Review by alxhub — APPROVED** [review]
Reviewed-For: global-approvers

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 4f7a838e9d0d1c59a18839ecc9b8db8e0927490e.

**Pull request closed without merging**